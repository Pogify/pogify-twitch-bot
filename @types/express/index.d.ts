declare namespace Express {
  interface Request {
    twitch: {
      broadcaster_type: BroadecasterType;
      description: string;
      display_name: string;
      email?: string;
      id: string;
      login: string;
      offline_image_url: string;
      profile_image_url: string;
      type: type;
      view_count: number;
      created_at: string;
    };
  }
}

type BroadecasterType = "partner" | "affiliate" | "";
type type = "staff" | "admin" | "global_mod" | "";
