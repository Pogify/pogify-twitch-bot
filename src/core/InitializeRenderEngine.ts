/* eslint-disable @typescript-eslint/no-explicit-any */
// @ts-expect-error || no types
import whiskers from "whiskers";
import path from "path";
import RendererConfig from "../../configs/RendererConfig.json";

interface IAppWithRenderer {
  engine(
    ext: string,
    renderer: (
      // eslint-disable-next-line no-shadow
      path: string,
      options: Record<string, any>,
      callback: (e: any, rendered?: string | undefined) => void
    ) => any
  ): any;
  set(setting: string, val: any): any;
}

export default function InitializeRenderEngine(app: IAppWithRenderer): void {
  // eslint-disable-next-line no-underscore-dangle
  app.engine(RendererConfig.extension, whiskers.__express);
  app.set("views", path.join(process.cwd(), RendererConfig.viewsFolder));
}
