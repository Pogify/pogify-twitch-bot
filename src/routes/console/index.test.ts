import request from "supertest";
import consoleRouter from ".";
import express from "express";
import InitializeRenderEngine from "../../core/InitializeRenderEngine";

let app: express.Express;
beforeEach(() => {
  app = express();
  InitializeRenderEngine(app);
});

test("test paths", async () => {
  app.use("/", consoleRouter);

  let res = await request(app).get("/");
  console.log(res.body);
  expect(res.status).toBe(200);
  expect(res.body).toMatchSnapshot();
});
