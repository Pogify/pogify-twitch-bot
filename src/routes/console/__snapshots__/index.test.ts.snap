// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test paths 1`] = `Object {}`;
